Use `api.name_converter` if exists in JsonLd\ContextBuilder & Hydra\ApiDocumentationBuilder

ogizanagi opened the pull request.

dunglas commented:
Can you add some tests?

ogizanagi commented:
Here tests are.

ogizanagi commented on features/json-ld/context.feature:
```diff
@@ -17,15 +17,15 @@ Feature: JSON-LD contexts generation
-            "relatedDummy": {
-                "@id": "#Dummy/relatedDummy",
+            "related_dummy": {
+                "@id": "#Dummy/related_dummy",
```
I have some doubts about those changes, as I don't use the converter for `@id`.

dunglas replied:
They are not OK. The key should be changed but not the value because the  referenced id is not touched: https://github.com/ogizanagi/DunglasApiBundle/blob/vocab_context_name_converter/features/hydra/vocab.feature#L105

dunglas commented on Hydra/ApiDocumentationBuilder.php:
```diff
@@ -134,10 +142,10 @@ public function getApiDocumentation()
                         '@id' => ($iri = $attributeMetadata->getIri()) ? $iri : sprintf('#%s/%s', $shortName, $attributeName),
                         '@type' => $type,
-                        'rdfs:label' => $attributeName,
+                        'rdfs:label' => $convertedName,
```
Does this change have an interest?

ogizanagi replied:
I only reflect what has been done in http://www.markus-lanthaler.com/hydra/api-demo/vocab (search for `vocab:User/raisedIssues`).
The `hydra:title` is used in HydraConsole to display the property in the documentation panel on the right, or the form labels when trying to make a POST operation for instance.

Also, there is a problem using NelmioApiDocBundle: the properties aren't converted with the converter neither.

What do you think ?

dunglas replied:
`rdf:label` and `hydra:title` are indeed descriptions for human readers. You can let it as is.

For Nelmio, it will need a patch in that bundle directly.

dunglas commented on Hydra/ApiDocumentationBuilder.php:
```diff
@@ -134,10 +142,10 @@ public function getApiDocumentation()
                         'domain' => $prefixedShortName,
                     ],
-                    'hydra:title' => $attributeName,
+                    'hydra:title' => $convertedName,
```
Same here.

dunglas commented:
I'm not confortable having the name converted blindly used on all tests. What do you think of that other strategy:
- create a new property "nameConverted" in an object
- create a custom name convert doing nothing for all properties except "nameConverted"

All tests will follow the standard and recommended convention except one.

ogizanagi commented:
I think it's a great idea, and I didn't though about it. I was also concerned about impacting the whole tests suite, but loading multiple configuration would have been overkill. So your solution is the best option :)

ogizanagi commented:
- Reverted `Hydra/ApiDocumentationBuilder.php`
- Change strategy in tests:
    - Added a `$nameConverted` property in `Dummy` entity
    - Added a CustomConverter extending `Symfony\Component\Serializer\NameConverter\CamelCaseToSnakeCaseNameConverter` that will only convert a property named `nameConverted`

dunglas commented:
Thank you. Run PHP CS Fixer and Travis will be green again.

dunglas commented:
Thank you @ogizanagi
